app/pages/_app.tsx

```diff
@@ -64,6 +65,10 @@ export default class App extends NextApp<AppProps> {
 
     return (
       <Sentry.ErrorBoundary fallback={ErrorFallback}>
+        <SessionTimeoutHandler
+          pageComponent={Component}
```

Comment:
removing this prop and checking for `Component.isAccessProtected` here will make it easier to extract into its own package

Reply:
we're relying on the component changing for the refresh condition on the `useEffect` call, we'd need a roundabout way of reloading this component if we were changing this
What I can suggest is to create an interface
```javascript
interface ISupportAccessProtection{
  isAccessProtected: boolean;
}
```
And we'd use that for the type of the component, instead of depending on the `CiipPageComponent` type.